core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -401,8 +402,18 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
             }
 
             serviceResponse = serviceResponse.recover(cause -> {
-                // Recover a failed response with the default exception handler.
-                return exceptionHandler.convert(reqCtx, cause);
+                // Recover a failed response with the exception handler.
+                final HttpResponse converted = exceptionHandler.convert(reqCtx, cause);
+                assert converted != null;
```

Comment:
How about falling back to the default converter when `null` is returned, with a one-time warning message?

Reply:
We are already applying `DefaultExceptionHandler` to handle `null`. 
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1729
And we actually recommend returning `null` so we probably don't need the warning message?
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/ExceptionHandler.java#L41